onedal/basic_statistics/incremental_basic_statistics.py

```diff
@@ -70,10 +70,27 @@ def __init__(self, result_options="all"):
         self._reset()
 
     def _reset(self):
+        self._need_to_finalize = False
         self._partial_result = self._get_backend(
             "basic_statistics", None, "partial_compute_result"
         )
 
+    def __getstate__(self):
+        """
+        Converts estimator's data to serializable format.
+        All tables contained in partial result are converted to np.arrays.
+        Notes
```

Comment:
I think the format requirement for rendering this docstring would be to have the title as "Note" and with a newline above it.

I'd also say that this could be better left as a comment than as a docstring, since `__getstate__` is a built-in method in python.

Reply:
I think it is better to keep as docstring since I have been requested to add info from the Note to documentation. Probably the best way to do it is usage of `automethod`. Let me know if you have any objections against it.

Reply:
Why would that have been requested in the first place? It is rather uncommon to have docstrings for built-ins like these, and methods with two underscores shouldn't show up in autoclass docs.

For example, scikit-learn for the most part doesn't put docstrings on their `__getstate__`/`__setstate__` methods:
https://github.com/search?q=repo%3Ascikit-learn%2Fscikit-learn%20__getstate__&type=code

Reply:
In that case, you might want to add that info as a "note" section in the class docstring, because it otherwise wouldn't reach the final documentation.

This is what the docs show for me after building them from your branch - you'll see that autoclass doesn't list double-underscore methods:
![image](https://github.com/user-attachments/assets/d916224d-5a6b-4c61-b149-b7c158628072)